lib/lambda_ethereum_consensus/state_transition/mutators.ex

```diff
@@ -160,4 +160,61 @@ defmodule LambdaEthereumConsensus.StateTransition.Mutators do
   defp whistleblower_index(whistleblower_index, proposer_index) do
     if whistleblower_index == nil, do: proposer_index, else: whistleblower_index
   end
+
+  @spec apply_deposit(
+          BeaconState.t(),
+          SszTypes.bls_pubkey(),
+          SszTypes.bytes32(),
+          SszTypes.uint64(),
+          SszTypes.bls_signature()
+        ) :: {:ok, BeaconState.t()} | {:error, binary()}
+  def apply_deposit(state, pubkey, withdrawal_credentials, amount, signature) do
+    validator_pubkeys = Enum.map(state.validators, fn validator -> validator.pubkey end)
+
+    if Enum.member?(validator_pubkeys, pubkey) do
+      index = Enum.find_index(validator_pubkeys, fn validator -> validator == pubkey end)
+      state = increase_balance(state, index, amount)
+      {:ok, state}
+    else
+      deposit_message = %SszTypes.DepositMessage{
+        pubkey: pubkey,
+        withdrawal_credentials: withdrawal_credentials,
+        amount: amount
+      }
+
+      domain = Misc.compute_domain(Constants.domain_deposit())
+
+      signing_root = Misc.compute_signing_root(deposit_message, domain)
+
+      case Bls.verify(pubkey, signing_root, signature) do
+        {:ok, verified} ->
+          apply_initial_deposit(verified, state, pubkey, withdrawal_credentials, amount)
+
+        {:error, msg} ->
+          {:error, msg}
+      end
```

Comment:
Whenever we have a case and we want to ignore the default case, let's use `with` instead:

```elixir
with {:ok, verified} <- Bls.verify(pubkey, signing_root, signature) do
  apply_initial_deposit(verified, state, pubkey, withdrawal_credentials, amount)
end
```

Reply:
does this also send the `{:error, msg}` in case it doesn't match in the examples i found that was like an else and sending `err -> err`

Reply:
Yes, it returns the error when it doesn't match. Doing `err -> err` is redundant UNLESS there there's a special error case being handled.